File: shared/aaaaaaaaaaaaaaa RLG Group Newsletter/rlg_group_newsletter.js
// RLG Group Automated Newsletter System
// Manages content creation, email campaigns, and trend-driven marketing automation.

import axios from "axios";
import OpenAI from "openai";
import nodemailer from "nodemailer";
import cron from "node-cron";
import cheerio from "cheerio";
import fs from "fs";

// ===== Configuration =====
const config = {
    openaiKey: process.env.OPENAI_API_KEY,
    deepseekKey: process.env.DEEPSEEK_API_KEY,
    emailSender: process.env.EMAIL_SENDER,
    emailPassword: process.env.EMAIL_PASSWORD,
    recipientsFile: "./subscribers.json",
    postFrequency: 4, // Number of posts per day
    newsletterSchedule: "0 10 * * *", // Send daily at 10 AM
    scrapeURLs: ["https://rlgdata.com", "https://rlgfans.com"]
};

// Initialize OpenAI Client
const openai = new OpenAI({
    apiKey: config.openaiKey
});

// Initialize Email Transporter
const transporter = nodemailer.createTransport({
    service: "gmail",
    auth: {
        user: config.emailSender,
        pass: config.emailPassword
    }
});

// ===== AI-Generated Content =====
async function generateNewsletterContent() {
    const prompt = `
    Generate a professional, engaging newsletter for RLG Data and RLG Fans.
    Include:
    - Industry trends in data intelligence and content monetization.
    - New RLG features & case studies.
    - Creator success stories.
    - A compelling call-to-action.
    `;
    
    try {
        const response = await openai.chat.completions.create({
            model: "gpt-4",
            messages: [{ role: "system", content: prompt }],
            max_tokens: 800
        });

        return response.choices[0].message.content;
    } catch (error) {
        console.error("Error generating newsletter content:", error);
        return "We encountered an issue generating this week's newsletter.";
    }
}

// ===== AI-Generated Images/Videos (OpenAI Sora) =====
async function generateMarketingMedia() {
    const prompt = "Create a high-quality ad image for RLG Data and RLG Fans featuring content analytics and creator monetization.";
    
    try {
        const response = await axios.post("https://api.openai.com/v1/images/generations", {
            model: "dall-e-3",
            prompt: prompt,
            n: 1,
            size: "1024x1024"
        }, {
            headers: { Authorization: `Bearer ${config.openaiKey}` }
        });

        return response.data.data[0].url;
    } catch (error) {
        console.error("Error generating marketing image:", error);
        return null;
    }
}

// ===== Email Sending =====
async function sendNewsletter() {
    const content = await generateNewsletterContent();
    const imageUrl = await generateMarketingMedia();
    const subscribers = JSON.parse(fs.readFileSync(config.recipientsFile, "utf8"));

    const emailOptions = {
        from: `"Khoto Zulu" <${config.emailSender}>`,
        subject: "🚀 RLG Data & RLG Fans - This Week’s Insights & Monetization Strategies",
        html: `
            <h2>📊 RLG Weekly Insights</h2>
            <p>${content}</p>
            <img src="${imageUrl}" alt="Marketing Image" style="max-width: 100%;">
            <p><a href="https://rlgdata.com/signup" style="padding: 10px; background: #007bff; color: white; text-decoration: none;">Join Now</a></p>
        `
    };

    for (const subscriber of subscribers) {
        emailOptions.to = subscriber.email;
        try {
            await transporter.sendMail(emailOptions);
            console.log(`Newsletter sent to: ${subscriber.email}`);
        } catch (error) {
            console.error(`Error sending to ${subscriber.email}:`, error);
        }
    }
}

// ===== Web Scraping for New Subscribers =====
async function scrapeEmails() {
    let newEmails = [];
    
    for (const url of config.scrapeURLs) {
        try {
            const { data } = await axios.get(url);
            const $ = cheerio.load(data);
            $("a[href^='mailto:']").each((_, el) => {
                const email = $(el).attr("href").replace("mailto:", "");
                if (email && !newEmails.includes(email)) {
                    newEmails.push(email);
                }
            });
        } catch (error) {
            console.error("Error scraping emails from:", url, error);
        }
    }

    if (newEmails.length > 0) {
        let existingEmails = JSON.parse(fs.readFileSync(config.recipientsFile, "utf8"));
        let updatedEmails = [...new Set([...existingEmails, ...newEmails])];
        fs.writeFileSync(config.recipientsFile, JSON.stringify(updatedEmails, null, 2));
        console.log("Updated subscriber list with new emails.");
    }
}

// ===== Social Media Auto-Posting =====
async function generateAndPostSocialMedia() {
    const topics = ["creator monetization", "AI-driven analytics", "trending insights", "content engagement"];
    const topic = topics[Math.floor(Math.random() * topics.length)];

    const prompt = `Create an engaging social media post about ${topic}, promoting RLG Data and RLG Fans.`;

    try {
        const response = await openai.chat.completions.create({
            model: "gpt-4",
            messages: [{ role: "system", content: prompt }],
            max_tokens: 200
        });

        const postContent = response.choices[0].message.content;
        console.log(`Generated Social Media Post: ${postContent}`);

        // Simulating post (replace with API calls to Twitter, LinkedIn, etc.)
        console.log("Auto-posting to social platforms...");
    } catch (error) {
        console.error("Error generating social media content:", error);
    }
}

// ===== Scheduling & Automation =====
cron.schedule(config.newsletterSchedule, async () => {
    console.log("Sending weekly newsletter...");
    await sendNewsletter();
});

cron.schedule("0 */6 * * *", async () => {
    console.log("Auto-posting marketing content...");
    await generateAndPostSocialMedia();
});

cron.schedule("0 0 * * 0", async () => {
    console.log("Scraping new email subscribers...");
    await scrapeEmails();
});

// ===== Execution =====
(async () => {
    console.log("🚀 RLG Marketing Agent Running...");
    await scrapeEmails();
    await generateAndPostSocialMedia();
})();

File: shared/RLG Group Newsletter/rlg_group_newsletter.py
# RLG_group_newsletter.py - AI-Powered Automated Newsletter for RLG Data & RLG Fans

import os
import json
import smtplib
import schedule
import time
import requests
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Configuration
CONFIG = {
    "openai_api_key": os.getenv("OPENAI_API_KEY"),
    "deepseek_api_key": os.getenv("DEEPSEEK_API_KEY"),
    "sora_api_key": os.getenv("SORA_API_KEY"),
    "email_sender": os.getenv("EMAIL_SENDER"),
    "email_password": os.getenv("EMAIL_PASSWORD"),
    "mailing_list_file": "./mailing_list.json",
    "newsletter_schedule": "Monday 09:00",  # Weekly newsletter every Monday at 9 AM
}

# AI Content Generator (ChatGPT & DeepSeek)
def generate_newsletter_content():
    """Generates newsletter content using AI models."""
    prompt = (
        "Write a professional, engaging, and informative newsletter for RLG Data & RLG Fans. "
        "Include insights on data analytics, monetization tips, trending strategies, "
        "and upcoming features. Ensure the content is user-friendly and relevant."
    )
    
    headers = {"Authorization": f"Bearer {CONFIG['deepseek_api_key']}"}
    
    try:
        response = requests.post(
            "https://api.deepseek.com/v1/chat/completions",
            json={"model": "deepseek-llm-1.3b", "messages": [{"role": "user", "content": prompt}], "max_tokens": 1000},
            headers=headers,
            timeout=10
        )
        
        return response.json()["choices"][0]["message"]["content"]
    
    except requests.RequestException as e:
        print(f"Error generating newsletter content: {e}")
        return "Error: Unable to generate newsletter content."

# AI Image & Video Generator (OpenAI Sora)
def generate_media(prompt, media_type="image"):
    """Generates AI-powered images or videos for newsletters using OpenAI Sora."""
    endpoint = "https://api.openai.com/v1/images" if media_type == "image" else "https://api.openai.com/v1/videos"
    
    headers = {"Authorization": f"Bearer {CONFIG['sora_api_key']}", "Content-Type": "application/json"}
    
    try:
        response = requests.post(endpoint, json={"prompt": prompt}, headers=headers, timeout=10)
        return response.json().get("url", "No media generated")
    
    except requests.RequestException as e:
        print(f"Error generating {media_type}: {e}")
        return None

# Email Sending Function
def send_newsletter():
    """Fetches mailing list and sends AI-generated newsletter to all subscribers."""
    
    # Load mailing list
    try:
        with open(CONFIG["mailing_list_file"], "r") as file:
            mailing_list = json.load(file)
    except (FileNotFoundError, json.JSONDecodeError):
        print("Error: Mailing list file not found or corrupted.")
        return
    
    if not mailing_list:
        print("No subscribers found in the mailing list.")
        return

    # Generate content and media
    newsletter_content = generate_newsletter_content()
    image_url = generate_media("Marketing image for RLG Data & RLG Fans.")
    
    # Email setup
    server = smtplib.SMTP_SSL("smtp.gmail.com", 465)
    server.login(CONFIG["email_sender"], CONFIG["email_password"])

    for subscriber in mailing_list:
        msg = MIMEMultipart()
        msg["From"] = CONFIG["email_sender"]
        msg["To"] = subscriber["email"]
        msg["Subject"] = f"🚀 Weekly Insights from RLG Data & RLG Fans - {time.strftime('%B %d, %Y')}"
        
        email_body = f"""
        <html>
        <body>
            <h2>Hello {subscriber['name']},</h2>
            <p>{newsletter_content}</p>
            <br>
            <p><strong>Featured Image:</strong></p>
            <img src="{image_url}" alt="RLG Data & RLG Fans">
            <br><br>
            <p>Stay ahead with RLG Data & RLG Fans! 🚀</p>
            <p>Best regards,<br><strong>Khoto Zulu</strong><br>Your AI Marketing Agent</p>
        </body>
        </html>
        """
        
        msg.attach(MIMEText(email_body, "html"))
        
        try:
            server.sendmail(CONFIG["email_sender"], subscriber["email"], msg.as_string())
            print(f"Newsletter sent to {subscriber['email']}")
        except smtplib.SMTPException as e:
            print(f"Error sending email to {subscriber['email']}: {e}")

    server.quit()

# Schedule Weekly Newsletters
schedule.every().monday.at(CONFIG["newsletter_schedule"].split()[1]).do(send_newsletter)

print("RLG Newsletter Agent is running...")

# Run scheduler continuously
while True:
    schedule.run_pending()
    time.sleep(60)
